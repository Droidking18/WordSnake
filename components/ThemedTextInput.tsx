import { TextInput, type TextProps, StyleSheet, useColorScheme } from 'react-native';

import { useThemeColor } from '@/hooks/useThemeColor';

export type ThemedTextInputProps = TextProps & {
  lightColor?: string;
  darkColor?: string;
  type?: 'default' | 'title' | 'defaultSemiBold' | 'subtitle' | 'link';
};

export function ThemedTextInput({
  style,
  lightColor,
  darkColor,
  type = 'default',
  ...rest
}: ThemedTextInputProps) {
  const color = useThemeColor({ light: lightColor, dark: darkColor }, 'text');
  // If we want border color to be themed as well, we can use useThemeColor for that too
  const isDarkMode = useColorScheme() === 'dark';

  return (
    <TextInput
      style={[
        { color },
        {
          borderColor: isDarkMode ? 'white' : 'black',
        },
        type === 'default' ? styles.default : undefined,
        type === 'title' ? styles.title : undefined,
        type === 'defaultSemiBold' ? styles.defaultSemiBold : undefined,
        type === 'subtitle' ? styles.subtitle : undefined,
        type === 'link' ? styles.link : undefined,
        style,
      ]}
      {...rest}
    />
  );
}

const styles = StyleSheet.create({
  default: {
    fontSize: 14,
  },
  defaultSemiBold: {
    fontSize: 16,
    lineHeight: 24,
    fontWeight: '600',
  },
  title: {
    fontSize: 32,
    fontWeight: 'bold',
    lineHeight: 32,
  },
  subtitle: {
    fontSize: 20,
    fontWeight: 'bold',
  },
  link: {
    lineHeight: 30,
    fontSize: 16,
    color: '#0a7ea4',
  },
});
