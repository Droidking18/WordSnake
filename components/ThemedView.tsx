import { ScrollView, View, type ViewProps } from 'react-native';

import { useThemeColor } from '@/hooks/useThemeColor';

export type ThemedViewProps = ViewProps & {
  lightColor?: string;
  darkColor?: string;
  scrollable?: boolean;
};

export function ThemedView({ style, lightColor, darkColor, ...otherProps }: ThemedViewProps) {
  const backgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, 'background');

  if (otherProps.scrollable == true) {
    return <ScrollView style={[{ backgroundColor }, style]} {...otherProps} />;
  } else {
    return <View style={[{ backgroundColor }, style]} {...otherProps} />;
  }
}
