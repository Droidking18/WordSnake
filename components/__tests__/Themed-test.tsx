import * as React from 'react';
import renderer from 'react-test-renderer';

import { ThemedText } from '../ThemedText';
import { ThemedTextInput } from '../ThemedTextInput';
import { ThemedView } from '../ThemedView';

import { Collapsible } from '../Collapsible';

it(`renders correctly`, () => {
  const tree = renderer.create(<ThemedText>Snapshot test!</ThemedText>).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`renders correctly`, () => {
  const tree = renderer.create(<ThemedTextInput />).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`renders correctly`, () => {
  const tree = renderer.create(<ThemedView />).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`renders correctly`, () => {
  const tree = renderer.create(<Collapsible title='Test'/>).toJSON();

  expect(tree).toMatchSnapshot();
});
