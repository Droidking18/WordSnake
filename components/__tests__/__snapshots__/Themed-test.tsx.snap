// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#11181C",
      },
      {
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Snapshot test!
</Text>
`;

exports[`renders correctly 2`] = `
<TextInput
  style={
    [
      {
        "color": "#11181C",
      },
      {
        "borderColor": "black",
      },
      {
        "fontSize": 14,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
/>
`;

exports[`renders correctly 3`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
/>
`;

exports[`renders correctly 4`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 6,
        "opacity": 1,
      }
    }
  >
    <Text />
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          undefined,
          undefined,
          {
            "fontSize": 16,
            "fontWeight": "600",
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      Test
    </Text>
  </View>
</View>
`;
